behavior/plans/diff.py

```diff
@@ -1,466 +1,285 @@
+    df[targets] : dataframe constructed from the CSV file for differencing
+    df[features] : dataframe constructed from the CSV file of non-differencing columns
+
+    Note:
+        - Both dataframes contain `ou_index` and `data_id` for joining the two dataframes
+        - Dataframe for differencing follows the COMMON_SCHEMA layout
     """
-    remapped = {}
-
-    for ou_name, df in ou_to_df.items():
-        remapper: dict[str, str] = {}
-
-        for init_col in df.columns:
-            found = False
-            for common_col in COMMON_SCHEMA:
-                if (
-                    common_col != init_col
-                    and common_col in init_col
-                    and init_col
-                    not in BASE_TARGET_COLS
-                    + [
-                        "left_child_plan_node_id",
-                        "right_child_plan_node_id",
-                        "plan_node_id",
-                    ]
-                ):
-                    assert not found, f"col: {init_col} and {common_col}"
-                    assert init_col not in remapper, remapper
+
+    # Read the CSV file and add `ou_index` and `data_id`.
+    df = pd.read_csv(csv_file, index_col=None)
+    df["ou_index"] = ou_index
+    df["data_id"] = df.index
+
+    # Determines which columns to remap. A non-target input column that is suffixed by a
+    # column in STANDARDIZE_COLUMNS is remapped to produce a common schema.
+    remapper: dict[str, str] = {}
+    for init_col in df.columns:
+        if init_col not in STANDARDIZE_COLUMNS and init_col not in BASE_TARGET_COLS:
+            for common_col in STANDARDIZE_COLUMNS:
+                if common_col in init_col:
                     remapper[init_col] = common_col
-                    found = True
+                    break
+    df.rename(columns=remapper, inplace=True)
+
+    # Here we produce two different dataframes from the original CSV file.
+    #
+    # df[targets] produces a dataframe containing all the common schema columns and the
+    # target columns. This dataframe is used to perform plan differencing on.
+    #
+    # df[features] contains all the other columns in the input data that were not used
+    # for differencing along with `ou_index` and `data_id`. It is worth noting that
+    # `ou_index` and `data_id` can be used to reconstruct a datapoint across the
+    # two dataframes.
+    targets = COMMON_SCHEMA + DIFF_COLS
+    features = ["ou_index", "data_id"] + list(set(df.columns) - set(targets))
+
+    # pylint: disable=E1136
+    return ou_index, df[targets], df[features]
+
+
+def diff_query_invocation(subinvocation):
+    """
+    Diffs a given query invocation by calling a CPython function.
+
+    Parameters
+    ----------
+    subinvocation : DataFrame describing a single invocation of an unique query instance
 
-        df = df.rename(columns=remapper)
+    Returns
+    -------
+    None : if the invocation cannot be diffed successfully.
```

Comment:
Please fix return docstring format.

Also, if you convert it to be exceptional control flow, you can get rid of the None case (what is the caller supposed to do with that?)

Reply:
The `None` case will still be needed since some error cases are not fatal (e.g., we lost events).